### PR TITLE
Ability to open spc from something that impl's Read + Seek

### LINKED ISSUE #5: Ability to load SPC files from something that impl's Read
It would be great if you could not just load from the file system, but also load from something that impl's the Read trait. This trait is e.g. used by reqwest for HTTP responses or also implemented for in-memory buffers. The audrey library operates on the Read trait as well, which means that this is required for audrey integration.

### PR DESCRIPTION
**Pull request by est31**
Fixes #5 .

### CODE REVIEW

**Comment by est31** [comment]
As the port is not done yet, I haven't verified whether there are any other things I need from the spc crate for audrey support, so please don't release a new version of it on crates.io just yet so that there can be further possible changes until such a release :).

File: src/spc.rs [hunk 1]
```diff
@@ -33,7 +33,10 @@ pub struct Spc {
 impl Spc {
     pub fn load<P: AsRef<Path>>(path: P) -> Result<Spc> {
         let file = try!(File::open(path));
-        let mut r = BinaryReader::new(BufReader::new(file));
+        Spc::from_reader(BufReader::new(file))
+    }
+    pub fn from_reader<R: Read + Seek>(reader: R) -> Result<Spc> {
```
Comment (yupferris) [inline src/spc.rs]: Small nit: mind adding a newline between these two fn definitions?

**Review by yupferris — APPROVED** [review]
Thanks so much for this! One small nit that I'd appreciate if you wouldn't mind fixing, otherwise I can merge as is. :)

**Comment by yupferris** [comment]
Also it's no problem waiting to release a new version; I can wait until you have what you need.

**Comment by est31** [comment]
@yupferris the PR is amended now :)

**Comment by yupferris** [comment]
Thanks :)

**Pull request merged**